crates/sui/src/client_commands.rs

```diff
@@ -2247,3 +2314,86 @@ pub async fn request_tokens_from_faucet(
     }
     Ok(())
 }
+
+fn pretty_print_balance(
+    coins: &BTreeMap<String, (Option<SuiCoinMetadata>, Vec<Coin>)>,
+    builder: &mut TableBuilder,
+    with_coins: &bool,
+) {
+    fn print(
+        builder: &mut TableBuilder,
+        coins: Vec<&(Option<SuiCoinMetadata>, Vec<Coin>)>,
+        with_coins: &bool,
+    ) {
+        let mut table_builder = TableBuilder::default();
+        for (metadata, coins) in coins {
+            let (name, symbol, decimals) = if let Some(metadata) = metadata {
+                (
+                    metadata.name.clone(),
+                    metadata.symbol.clone(),
+                    metadata.decimals,
+                )
+            } else {
+                ("unknown".to_string(), "unknown symbol".to_string(), 10u8)
+            };
```

Comment:
Do these need to be `String`s, or could they just be `&str`s which would avoid the clones?
```suggestion
            let (name, symbol, decimals) = if let Some(metadata) = &metadata {
                (
                    metadata.name.as_str(),
                    metadata.symbol.as_str(),
                    metadata.decimals,
                )
            } else {
                ("unknown", "unknown symbol", 10u8)
            };
```

Reply:
I'll check.

Reply:
Switched over to @amnn's suggestion of using `entry` api, so it should be fine.

Reply:
Indeed we can.